src/Momento.Sdk/Internal/Utils.cs

```diff
@@ -199,4 +194,15 @@ public static void ValidateStartEndIndex(int? startIndex, int? endIndex)
         }
         return;
     }
+
+    /// <summary>
+    /// Calculate the deadline given a timeout.
+    /// </summary>
+    /// <param name="timeout"></param>
+    /// <returns></returns>
+    public static DateTime CalculateDeadline(TimeSpan timeout)
+    {
+        return DateTime.UtcNow.Add(timeout);
+    }
```

Comment:
I think there's another `CalculateDeadline` implementation in the code elsewhere. Could make that code use this version and delete the other implementation.